src/quacc/calculators/espresso/calculators.py

```diff
@@ -0,0 +1,127 @@
+from quacc import SETTINGS
+from quacc.calculators.espresso.core import EspressoCalculator
+from quacc.calculators.espresso.keys import (DYNMAT_KEYS, NEB_KEYS, PH_KEYS,
+                                             Q2R_KEYS, MATDYN_KEYS, PW_KEYS)
+from quacc.calculators.espresso.utils import parse_pw_preset
+
+# Mini classes that might be useful for binaries
+# that have very specific requirements
+
+class Ph(EspressoCalculator):
+    # These are not supposed to change
+    _BIN = 'ph'
+    _KEYS = PH_KEYS
+    _CMD = SETTINGS['ESPRESSO_PH_CMD']
+
+    def __init__(self, input_data):
+        self.input_data = input_data
+        super().__init__()
+
+class Neb(EspressoCalculator):
+
+    # I think it makes sense to have the defaults
+    # here. They are not supposed to change.
+    # Users can have a look at them by doing:
+    # from quacc.calculators.espresso.calculators import Neb
+    # Neb._calc_defaults without looking at the source code
+
+    # Note to self: 'When a child class method is created with the same name and signature as one in the parent, the child's method takes precedence' didn't know that, praise python
+    _BIN = 'neb'
+    _KEYS = NEB_KEYS
+    _CMD = SETTINGS['ESPRESSO_NEB_CMD']
+
+    _calc_defaults = {
+        'input_data': {
+            'path': {
+                'string_method': 'neb',
+                'restart_mode': 'from_scratch',
+                'opt_scheme': 'broyden',
+                'CI_scheme': 'no-CI',
+                'first_last_opt': False,
+                "num_of_images": 6
+            }
+        }
+    }
+
+    def __init__(self, neb_input_data, pw_input_data):
+        self.input_data = neb_input_data
+        self.additional_data = pw_input_data
+        super().__init__()
+        self.input_file = 'neb.dat'
+
+    # It would be nice to be able to run sanity_checks BEFORE
+    # the calculation is submitted to any kind of HPC
+    # This could also raise warnings to remind the user
+    # about known problems i.e. if first_last_opt is not True
+    # and the first and last images contains spin state. Running
+    # the first singlepoint with fresh wavefunctions might end up
+    # in a different spin state, screwing up the whole NEB calculation.
+
+    # Probably the best way is to have a recipe that only does sanity checks
+    # with a different runner (local) and then the actual calculation with
+    # the HPC runner. This way we can also check if the calculation is
+    # going to fail.
```

Comment:
If I'm understanding your comment correctly, I'll note that this is something I have not yet fully figured out how to do smoothly in a programmatic way.

In principle, if quacc _only_ supported Covalent (for instance), one could do a `@ct.electron(executor="local")` around the relevant function that is basically no-cost. However, because quacc supports several workflow engines out of the box and they all specify executors in slightly different ways, there is no clean mechanism for doing this at the moment (since the kwargs of the `@job` decorator will differ for each workflow engine).

There are probably some ideas I can pull from the [psiflow](https://github.com/molmod/psiflow) project, which uses a YAML file to have the user define multiple configurations (e.g. one for local, one for simulations, one for ML) that could be pulled from, but that's not something I've had time to try to sort out.

I should note that simply not including a decorator at all would make it run locally, but this isn't viable because the inputs to the function might be a `Future` from another function call, which would only be resolved if you were using a `@job`.

Reply:
The burden of running it locally will be on the users, so it will be a possibility for them, but yes I don't think there is anyway to enforce that directly in quacc